### PR TITLE
Set expectation criteria to use leaseKey

### PR DESCRIPTION
**Pull request by Renjuju**
## Changes
* Enforces updates to lease entry to currently exist + have the lease key field

## Test results
image

### CODE REVIEW

**Review by ychunxue — APPROVED** [review]
LGTM

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseSerializer.java [hunk 1]
```diff
@@ -193,10 +193,12 @@ public Lease fromDynamoRecord(Map<String, AttributeValue> dynamoRecord, Lease le
     }
 
     @Override
-    public Map<String, ExpectedAttributeValue> getDynamoExistantExpectation() {
+    public Map<String, ExpectedAttributeValue> getDynamoExistentExpectation(String leaseKey) {
```
Comment (ashwing) [inline amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseSerializer.java:196]: Can we add an integration test in DynamoDBLeaseRefresherIntegrationTest for the method updateLeaseWithMetaInfo ?

Comment (Renjuju) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseSerializer.java:196]: added integration tests

**Review by ashwing — APPROVED** [review]
Nice! LGTM.

**Pull request merged**